modules/swagger-codegen/src/main/resources/asyncscala/api.mustache

```diff
@@ -40,9 +40,14 @@ class {{classname}}(client: TransportClient, config: SwaggerConfig) extends ApiC
     {{/required}}
     {{/queryParams}}
 
-    {{#headerParams}}
-    headerParams += "{{baseName}}" -> {{paramName}}.toString
-    {{/headerParams}}
+  {{#headerParams}}
+      {{^required}}
+          if ({{paramName}} != null) {{paramName}}.map { v => headerParams += "{{baseName}}" -> v.toString }
```

Comment:
Can we use 2-space indentation instead?

And what if {{paramName}} (variable) is a non-string? Shall we use {{#isString}} ... {{/isString}} to do the null check only if it's a string?